### PR TITLE
Inception Score

### PR DESCRIPTION
**Pull request by SkafteNicki**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
Adds Inception Score (IS) to the new image module
Also added corrected a bit about the FID metric

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/image/test_inception.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+
+def test_is_raises_errors_and_warnings():
+    """ Test that expected warnings and errors are raised """
+    with pytest.warns(
+        UserWarning,
+        match='Metric `IS` will save all extracted features in buffer.'
+        ' For large datasets this may lead to large memory footprint.'
+    ):
+        _ = IS()
```
Comment (Borda) [inline tests/image/test_inception.py:63]:
```suggestion
        IS()
```

File: tests/image/test_inception.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        with pytest.raises(ValueError, match='Integer input to argument `feature` must be one of .*'):
+            _ = IS(feature=2)
+    else:
+        with pytest.raises(
+            ValueError,
+            match='IS metric requires that Torch-fidelity is installed.'
+            'Either install as `pip install torchmetrics[image-quality]`'
+            ' or `pip install torch-fidelity`'
+        ):
+            _ = IS()
```
Comment (Borda) [inline tests/image/test_inception.py:75]:
```suggestion
            IS()
```

File: tests/image/test_inception.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+    with pytest.raises(ValueError, match='Got unknown input to argument `feature`'):
+        _ = IS(feature=[1, 2])
```
Comment (Borda) [inline tests/image/test_inception.py:78]:
```suggestion
        IS(feature=[1, 2])
```

File: torchmetrics/image/inception.py [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+        rank_zero_warn(
+            'Metric `IS` will save all extracted features in buffer.'
+            ' For large datasets this may lead to large memory footprint.', UserWarning
+        )
+
+        if isinstance(feature, (str, int)):
+            if not _TORCH_FIDELITY_AVAILABLE:
+                raise ValueError(
+                    'IS metric requires that Torch-fidelity is installed.'
+                    'Either install as `pip install torchmetrics[image-quality]`'
```
Comment (Borda) [inline torchmetrics/image/inception.py:125]:
```suggestion
                    'Either install as `pip install torchmetrics[image]`'
```
see: https://github.com/PyTorchLightning/metrics/blob/0ad5db63e99dbda32b0e75f666ae77efc71ad1de/setup.py#L29

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
@SkafteNicki seems they changed API in the newly release 0.3, renamed `input_1` to `input1` etc.
https://dev.azure.com/PytorchLightning/Metrics/_build/results?buildId=13410&view=logs&j=3afc50db-e620-5b81-6016-870a6976ad29&t=98354d77-e326-51ec-536f-1549451db1fa&l=7268

**Comment by SkafteNicki** [comment]
@Borda should have been fixed in commit 162d34081b0a4764839bee7f7c82dcce8fd0f72c

**Comment by Borda** [comment]
> @Borda should have been fixed in commit 162d34081b0a4764839bee7f7c82dcce8fd0f72c

Ok, but would work for v0.2

File: torchmetrics/image/inception.py [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+    """
+
+    def __init__(
+        self,
+        feature: Union[str, int, torch.nn.Module] = 'logits_unbiased',
+        splits: int = 10,
+        compute_on_step: bool = False,
+        dist_sync_on_step: bool = False,
+        process_group: Optional[Any] = None,
+        dist_sync_fn: Callable = None
```
Comment (justusschock) [inline torchmetrics/image/inception.py:107]: Nit: Can we specify the type of the Callable here? For str as input and int as output this would be `Callable[[str], int]`

File: torchmetrics/image/inception.py [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+                    ' or `pip install torch-fidelity`'
+                )
+            valid_int_input = ['logits_unbiased', 64, 192, 768, 2048]
+            if feature not in valid_int_input:
+                raise ValueError(
+                    f'Integer input to argument `feature` must be one of {valid_int_input},'
+                    f' but got {feature}.'
+                )
+
+            self.inception = NoTrainInceptionV3(name='inception-v3-compat', features_list=[str(feature)])
```
Comment (justusschock) [inline torchmetrics/image/inception.py:135]: Does it make sense to allow mutiple features here?

File: torchmetrics/image/inception.py [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+        )
+
+        if isinstance(feature, (str, int)):
+            if not _TORCH_FIDELITY_AVAILABLE:
+                raise ValueError(
+                    'IS metric requires that Torch-fidelity is installed.'
+                    'Either install as `pip install torchmetrics[image]`'
+                    ' or `pip install torch-fidelity`'
+                )
+            valid_int_input = ['logits_unbiased', 64, 192, 768, 2048]
```
Comment (justusschock) [inline torchmetrics/image/inception.py:128]:
```suggestion
            valid_int_input = ('logits_unbiased', 64, 192, 768, 2048)
```
use immutables if possible

File: torchmetrics/image/inception.py [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+                raise ValueError(
+                    f'Integer input to argument `feature` must be one of {valid_int_input},'
+                    f' but got {feature}.'
+                )
+
+            self.inception = NoTrainInceptionV3(name='inception-v3-compat', features_list=[str(feature)])
+        elif isinstance(feature, torch.nn.Module):
+            self.inception = feature
+        else:
+            raise ValueError('Got unknown input to argument `feature`')
```
Comment (justusschock) [inline torchmetrics/image/inception.py:139]:
```suggestion
            raise TypeError('Got unknown input to argument `feature`')
```

also reflect this in docstrings :)

**Review by justusschock — APPROVED** [review]
Some really minor comments. looks good to me

File: torchmetrics/image/inception.py (hunk 6 above)
Comment (SkafteNicki) [reply torchmetrics/image/inception.py:137]: Maybe, I do not see a problem with wanting to calculate the score for different features.
But IMO we should wait implementing that until someone has a specific usecase for wanting to do that.
